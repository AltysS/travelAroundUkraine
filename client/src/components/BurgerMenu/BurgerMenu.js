import React, { useState } from 'react';
import Box from '@mui/material/Box';
import IconButton from '@mui/material/IconButton';
import Menu from '@mui/material/Menu';
import MenuIcon from '@mui/icons-material/Menu';
import MenuItem from '@mui/material/MenuItem';
import { NavLink } from 'react-router-dom';
import appliedTheme from '../../theme/theme';

const pages = [
  {
    name: 'Home',
    link: '/',
    innerContent: 'Home',
  },
  {
    name: 'Catalogue',
    link: '/catalogue',
    innerContent: 'Catalogue',
  },
  {
    name: 'Cart',
    link: '/cart',
    innerContent: 'Cart',
  },
];

const styleNavLink = {
  display: 'block',
  textDecoration: 'none',
  color: appliedTheme.palette.text.primary,
  width: '100%',
  padding: '6px 16px',
};

const BurgerMenu = () => {
  const [anchorElNav, setAnchorElNav] = useState(null);

  const handleOpenNavMenu = (event) => {
    setAnchorElNav(event.currentTarget);
  };

  const handleCloseNavMenu = () => {
    setAnchorElNav(null);
  };

  const boxSX = {
    flexGrow: 1,
    display: { xs: 'flex', laptop: 'none' },
    justifyContent: 'flex-end',
  };

  const menuSX = {
    display: { xs: 'flex', laptop: 'none' },
  };

  const menuItemSX = {
    padding: 0,
    width: '40vw',
    maxWidth: '200px',
    height: '50px',
  };

  const anchorOrigin = {
    vertical: 'bottom',
    horizontal: 'left',
  };

  const transformOrigin = {
    vertical: 'top',
    horizontal: 'left',
  };

  return (
    <Box sx={boxSX}>
      <IconButton
        size="large"
        aria-label="appbar"
        aria-controls="menu-appbar"
        aria-haspopup="true"
        onClick={handleOpenNavMenu}
        sx={{ padding: 0, color: appliedTheme.palette.text.primary }}
      >
        <MenuIcon />
      </IconButton>

      <Menu
        id="menu-appbar"
        anchorEl={anchorElNav}
        anchorOrigin={anchorOrigin}
        keepMounted
        transformOrigin={transformOrigin}
        open={!!anchorElNav}
        onClose={handleCloseNavMenu}
        sx={menuSX}
      >
        {pages.map(({ name, link, innerContent }) => (
          <MenuItem key={name} onClick={handleCloseNavMenu} sx={menuItemSX}>
            <NavLink to={link} key={name} onClick={handleCloseNavMenu} style={styleNavLink}>
              {innerContent}
            </NavLink>
          </MenuItem>
        ))}
      </Menu>
    </Box>
  );
};

export default BurgerMenu;
