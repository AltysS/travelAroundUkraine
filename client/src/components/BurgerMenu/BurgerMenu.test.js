import { render } from '@testing-library/react';
import BurgerMenu from './BurgerMenu';
import { BrowserRouter } from 'react-router-dom';

jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/MenuItem', () => ({ children }) => <li>{children}</li>);
jest.mock('@mui/material/Menu', () => ({ children }) => <ul>{children}</ul>);
jest.mock('@mui/material/IconButton', () => ({ children }) => <button>{children}</button>);
jest.mock('@mui/material/Typography', () => ({ children }) => <a>{children}</a>);

describe('Render BurgerMenu', () => {
  test('should BurgerMenu render', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <BurgerMenu />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
