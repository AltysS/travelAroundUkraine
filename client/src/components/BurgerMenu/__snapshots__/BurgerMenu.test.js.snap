// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BurgerMenu should BurgerMenu render 1`] = `
<DocumentFragment>
  <div>
    <button>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
    <ul>
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
          style="display: block; text-decoration: none; color: rgb(40, 40, 40); width: 100%; padding: 6px 16px;"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class=""
          href="/catalogue"
          style="display: block; text-decoration: none; color: rgb(40, 40, 40); width: 100%; padding: 6px 16px;"
        >
          Catalogue
        </a>
      </li>
      <li>
        <a
          class=""
          href="/cart"
          style="display: block; text-decoration: none; color: rgb(40, 40, 40); width: 100%; padding: 6px 16px;"
        >
          Cart
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
